Build continuous integration (CI) workflows by using GitHub Actions

a-a-ron opened the pull request:
## Module: Build continuous integration (CI) workflows by using GitHub Actions

**Issue reference:** https://github.com/githubpartners/cs-public-content/issues/80#issuecomment-1184824007

### Module layout

![Screen Shot 2022-08-11 at 8 24 26 AM](https://user-images.githubusercontent.com/6351798/184156517-510ae7ab-3086-401b-ac3c-2ebb8ef6d38d.png)

a-a-ron committed:
Update index.yml

rmallorybpc committed:
Update 2-github-actions-workflows-ci.md

rmallorybpc committed:
Update 2b-github-actions-essential-features.md

rmallorybpc committed:
Merge pull request #15 from githubpartners/rmallorybpc-patch-10

Update 2b-github-actions-essential-features.md

rmallorybpc committed:
Merge pull request #14 from githubpartners/rmallorybpc-patch-9

Update 2-github-actions-workflows-ci.md

rmallorybpc committed:
Update 2-github-actions-workflows-ci.md

a-a-ron commented on github/github-actions-ci/includes/2-github-actions-workflows-ci.md:
````diff
@@ -177,6 +195,18 @@ Another action we could take is to add a label to the pull request. In this case
-Notice the block called ```env:```. This is where you set the environment variables for this action. For example, you can set the number of approvers needed. Here, it's one. The ```GITHUB_TOKEN``` variable is required because the action must make changes to your repository by adding a label. Finally, you supply the name of the label to add.
+Notice the block called ```env:```. This is where you set the environment variables for this action. For example, you can set the number of approvers needed. Here, it's one. The ```GITHUB_TOKEN``` variable is required because the action must make changes to your repository by adding a label. GitHub Actions lets you control the permissions granted to the GITHUB_TOKEN secret. A `permissions` key su ...
+
+GitHub actions now requires a collaborator with write access to take action to run workflows from first time users in pull requests.
````
@rmallorybpc I don't think this line fits in this section. I think it would be better to include this information in a section that talks about running workflows instead of automating reviews.

rmallorybpc replied:
@a-a-ron ok, I added the below section in [Build and deploy applications to Azure by using GitHub Actions](https://github.com/githubpartners/microsoft-learn/pull/6/files):
GitHub Actions lets you control the permissions granted to the `GITHUB_TOKEN` secret. The `GITHUB_TOKEN` is an automatically generated secret letting you make authenticated calls to the GitHub API in your workflow runs. Actions generates a new token for each job and expires the token when a job completes. These new settings allow you to follow a principle of least privilege in your workflows. The token has `write` permissions to a number of API endpoints except in the case of pull requests from forks which are always `read`.

a-a-ron replied:
I spent some time looking into this and I think we can add a new "short" section to add clarity around this statement. I really like the blog post about it: https://github.blog/2021-04-22-github-actions-update-helping-maintainers-combat-bad-actors/

Specifically the second feature in this section: https://github.blog/2021-04-22-github-actions-update-helping-maintainers-combat-bad-actors/#new-features-to-help-protect-maintainers

We don't really have an actions admin module that this would fit into other than admin for actions enterprise, but I don't really want to put it there since it's not enterprise specific. For now, I'm thinking here: https://docs.microsoft.com/en-us/learn/modules/github-actions-automate-tasks/2c-configure-github-actions-workflow

rmallorybpc replied:
Good morning @a-a-ron I added content, created a new branch, and tagged you for review.

rmallorybpc committed:
Update 2-github-actions-workflows-ci.md

GitHub Actions: Save time with partial re-runs in GitHub Actions #23 updates.

rmallorybpc committed:
Update 2-github-actions-workflows-ci.md

GitHub Actions: Re-run jobs with debug logging #21 updates.

a-a-ron commented:
@rmallorybpc, let's pause the work on this for a minute so we can discuss it during our Wednesday call. I think we may need to remodel this module. See here for more info: https://github.com/githubpartners/microsoft-learn/issues/28#issuecomment-1225006243

rmallorybpc committed:
Update 2b-github-actions-essential-features.md

Customize your workflow with artifact data and environment variables

Content from this unit in the Actions CI module
Removed the section, Pass artifact data between jobs, from microsoft-learn/github/github-actions-ci/includes/2b-github-actions-essential-features.md in github-actions-ci
Removed the section, Custom environment variables, from microsoft-learn/github/github-actions-ci/includes/2b-github-actions-essential-features.md in github-actions-ci

rmallorybpc committed:
Update 1-introduction.md

Added transition to next unit sentence.

rmallorybpc committed:
Update 2-github-actions-workflows-ci.md

Added transition to next unit sentence.

rmallorybpc committed:
Update 2b-github-actions-essential-features.md

Added transition to next unit sentence.

rmallorybpc committed:
Update 1-introduction.md

Edit updates after Cami's review.

rmallorybpc committed:
Update 2-github-actions-workflows-ci.md

Edit updates after Cami's review.

rmallorybpc committed:
Update 2b-github-actions-essential-features.md

Edit updates after Cami's review.

rmallorybpc committed:
Update 5-summary.md

Edit updates after Cami's review.
